lib/spack/spack/compilers/oneapi.py

```diff
@@ -131,11 +131,11 @@ def setup_custom_environment(self, pkg, env):
         # Edge cases for Intel's oneAPI compilers when using the legacy classic compilers:
         # Always pass flags to disable deprecation warnings, since these warnings can
         # confuse tools that parse the output of compiler commands (e.g. version checks).
-        if self.cc and self.cc.endswith("icc") and self.real_version >= Version("2021"):
+        # Update as per https://github.com/spack/spack/issues/47045
+        if self.real_version >= Version("2021") and self.real_version <= Version("2023"):
             env.append_flags("SPACK_ALWAYS_CFLAGS", "-diag-disable=10441")
-        if self.cxx and self.cxx.endswith("icpc") and self.real_version >= Version("2021"):
             env.append_flags("SPACK_ALWAYS_CXXFLAGS", "-diag-disable=10441")
-        if self.fc and self.fc.endswith("ifort") and self.real_version >= Version("2021"):
+        if self.real_version >= Version("2021") and self.real_version <= Version("2024"):
             env.append_flags("SPACK_ALWAYS_FFLAGS", "-diag-disable=10448")
```

Comment:
Regarding https://github.com/spack/spack/pull/47046#issuecomment-2435055348

> ... Therefore, we really only need the Fortran part of the logic in the oneAPI case.

My request is to remove the pointer

> Update as per https://github.com/spack/spack/issues/47045

in each class, and replace it with an explanation (in the oneapi case, about how this is probably only needed for ifort, perhaps the `%intel` case doesn't require additional explanation, since I don't think users would register `icx` as `intel`, but if you do think more explanation is necessary in that case, if you could expand it there vs. only referencing the link, that would be great.

Reply:
Done, please check